### DISCUSSION TITLE
#30DaysOfMSGraph (✅ Completed) - Top Microsoft Graph Scenarios

### DISCUSSION THREAD

**Discussion by japhletnwamu** [microsoft/30daysof#88] [Ideas]
### **Top Microsoft Graph Scenarios** 

**We want you to build amazing solutions with Microsoft Graph. In this discussion forum, we have provided some Microsoft Graph projects to serve as inspiration to you.** 

- Get upcoming meetings for the rest of the day 
- Send somebody an email when someone comes online 
- Send an email to the newly added user that was added to Azure Active Directory 
- Get students/people in my class/team 
- Get students/people from another class/team 
- Send a reminder for upcoming task on To-Do list 
- Summarize my last email (Use Cognitive Services for the summarization) 
- Get a Teams meeting call recording and send an e-mail to meeting participants 
- Setup a Teams meeting and return the URL 
- Send a message to a Teams channel or chat 
- Get a Teams meeting recording and summarize it (Use Cognitive Services for the summarization) 
- Summarize an email (Use Cognitive Services for the summarization) and send the summary to a Teams channel
- Retrieve files from OneDrive for Business and filter them based upon a specific company/teacher/university
- Retrieve emails and filter them based upon a specific company/teacher/university
- Retrieve Teams chat and filter them based upon a specific company/teacher/university
- Bot that provides you an auto suggestion of the recipient time zone (like day, night, already in a meeting, OOO), which avoids the time zone conversion while scheduling meetings
- Create a Microsoft 365 Group using custom resource behavior options
- Add a Microsoft Team to an existing Microsoft 365 Group
- Install a Microsoft Teams app as a personal app for individual users
- Archive Microsoft Teams
- Create personal tasks in Microsoft To-Do
- Create team tasks in Microsoft Planner
- Read and write to a OneNote notebook
- Read and write data to Excel workbooks
- Use graph to query a user’s colleagues (get manager and their direct report without the user) - We did this for our sample in learn together 😊
- Bot to show weather for the location the user is in. With command “Today’s weather”. Use city
- Use graph’s employeeHireDate data to send kudos on colleagues’ work anniversaries

**Comment by aycabas** [comment]
Please submit your project proposal below :) Feel free to pick one from above!

**Comment by NikitaChernevskiy** [comment]
My project will be an application in Power Pages, where will be 2 roles - employee and manager. Employee can see his/her tasks on To Do, see completed, late uncompleted and in progress tasks, calendar with his/her meetings and status online/offline of teammates. Manager can see all calendars of employees, create meetings with them in Microsoft teams, see activity status, see charts with completed, late uncompleted and in progress tasks for every single employee and create/delete tasks in To Do. The application will send also reminders about tasks, create meetings in calendar and so on. I will use Power Pages, Microsoft graph, Microsoft teams, Power Automate, Outlook and event hub.

**Reply by aycabas** [reply]
Great scenario Nikita, looking forward to hearing more about it! Let us know if you need any help 👏 If you successfully complete this, we'll invite you to speak with us on Microsoft 365 Developer YouTube channel for an interview, super excited for this! 🥰

**Reply by NikitaChernevskiy** [reply]
Thanks!

**Comment by sabrinadirnboeck** [comment]
The project I will be working on is a Microsoft Teams integrated app for placing orders for a canteen.
As data source sharepoint lists are going to be used.
The app should allow a user to select a menue and timeslot for the coming week and then save it to a sharpoint list. It should serve to replace excel sheets for ordering.
Furthermore there should be a functionality that allows to plan the selected menues with selected timeslot into the user's caldender.

**Comment by aborse555** [comment]
For my project, I will make a simple weather application that shows the current weather for the given location of the user. It should also be able to search for other locations as well.

**Reply by aycabas** [reply]
Hi @aborse555, great idea! Which part will you use Microsoft Graph for? Will you get user location through user profile with MS Graph?

**Reply by aborse555** [reply]
Thank you @aycabas ! Yes, I should have mentioned that, thank you!!

**Comment by PaMi40** [comment]
The submission is done if I just write here my proposal?

**Reply by aycabas** [reply]
Hi @PaMi40, you can write your proposal here :)

**Comment by SyedKazimRaza1** [comment]
I want to develop a gaming application in Microsoft teams, so that if user are feeling tired after work, they can play games with their team members and colleagues.

**Comment by balaupp** [comment]
My project would be to Send a message to a Teams channel or chat

**Reply by Edexde** [reply]
My project will be about Getting upcoming meetings for the rest of the day and display To-Do tasks as well

**Comment by corriespondent** [comment]
I'm going to try to look for available meeting times with colleagues using the findMeetingTimes endpoint. I'd like to hook in two different tenants (assuming that I have the ability to authenticate to both) but I'm not sure where to start with that -- anyone know of some reference links/docs/samples along those lines?

**Comment by PaMi40** [comment]
I would like to play with presence in my project.

**Comment by PaMi40** [comment]
Has anyone succeeded with calling the presence?

Which URL should I use?
https://graph.microsoft.com/v1.0/me/presence ???

**Reply by Edexde** [reply]
it works perfectly fine 
make sure you consent to the proper permissions when making the call
